internal/app/machined/pkg/controllers/secrets/api.go

```diff
@@ -103,16 +103,17 @@ func (ctrl *APIController) Run(ctx context.Context, r controller.Runtime, logger
 		}
 
 		// machine type is known and network is ready, we can now proceed to one or another reconcile loop
-		switch machineType {
+		switch machineType { //nolint:exhaustive
 		case machine.TypeInit, machine.TypeControlPlane:
 			if err = ctrl.reconcile(ctx, r, logger, true); err != nil {
 				return err
 			}
-		case machine.TypeJoin:
+		case machine.TypeWorker:
 			if err = ctrl.reconcile(ctx, r, logger, false); err != nil {
 				return err
 			}
-		case machine.TypeUnknown: // nothing to do
+		default:
+			panic(fmt.Sprintf("unexpected machine type %v", machineType))
```

Comment:
this panic is a bit too much here: `MachineType` resource initially is set to `machine.TypeUnknown` (until the machine config is loaded), so this branch is reachable